src/main/java/webserver/controllers/TodoController.java

```diff
@@ -98,10 +99,70 @@ public TodoController(TodoService todoService) {
             .build();
     };
 
+    public Callback<Request, String> getTodoDetailEdit = (request) -> {
+        final int id = getTodoIdFromPath(request.getPath());
+        String html = "";
+        int responseCode = 0;
+
+        Mustache template = MustacheUtil.getTemplate("todo-item-edit.mustache");
+        List<Todo> allTodos = todoService.getTodos();
+        Todo todo = TodosHelper.getTodo(id, allTodos);
+
+        if (todo == null) {
+            html = getTodoNotFound.apply(request);
+            responseCode = HTTP_HEADERS.STATUS_400;
+        } else {
+            html = MustacheUtil.executeTemplate(template, todo);
+            responseCode = HTTP_HEADERS.STATUS_200;
+        }
+
+        return new Response.Builder(responseCode)
+            .withHeader(HTTP_HEADERS.CONTENT_TYPE, "text/html; charset=utf-8")
+            .withHeader(HTTP_HEADERS.CONTENT_LENGTH, getContentLength(html))
+            .withBody(html)
+            .build();
+    };
+
+    public Callback<Request, String> editTodo = (request) -> {
+        int id = getTodoIdFromPath(request.getPath());
+        String contentType = request.getHeaders().get(HTTP_HEADERS.CONTENT_TYPE);
+        String body = request.getBody();
+        List<Todo> allTodos = todoService.getTodos();
+        Todo todo = TodosHelper.getTodo(id, allTodos);
+ 
+        if (contentType.equals("application/x-www-form-urlencoded")) {
```

Comment:
Pro-tip: It can come in handy to swap the order of the two strings.  `contentType` could be null if the header isn't set at all, which would cause `contentType.equals` to blow up.  But the reverse won't ever cause a null pointer exception, since it's a String constant

```java
if("application/x-www-form-urlencoded".equals(contentType)) ...
```